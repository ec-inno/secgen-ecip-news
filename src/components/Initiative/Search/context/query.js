import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
