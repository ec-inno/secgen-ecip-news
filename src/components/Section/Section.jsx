import React from 'react';
import PropTypes from 'prop-types';

const Section = ({ title, children }) => {
  if (!children) return '';

  return (
    <>
      {title && <h2 className="ecl-u-type-heading-2">{title}</h2>}
      {children}
    </>
  );
};

Section.propTypes = {
  title: PropTypes.string,
  children: PropTypes.any,
};

export default Section;
