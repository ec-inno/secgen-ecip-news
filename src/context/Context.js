import React from 'react';

const initialState = {
  hideOverlay: true,
};

export const SET_LANGUAGE_OVERLAY_VISIBILITY =
  'SET_LANGUAGE_OVERLAY_VISIBILITY';

const reducer = (state, action) => {
  switch (action.type) {
    case SET_LANGUAGE_OVERLAY_VISIBILITY: {
      return Object.assign({}, state, {
        hideOverlay: action.hideOverlay,
      });
    }

    default: {
      return initialState;
    }
  }
};

const Context = React.createContext();

const ContextProviderWithReducer = ({ children }) => {
  const [store, dispatch] = React.useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ store, dispatch }}>{children}</Context.Provider>
  );
};

export { ContextProviderWithReducer, Context as default };
