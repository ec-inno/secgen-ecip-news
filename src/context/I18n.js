import React from 'react';

const I18n = React.createContext();

export default I18n;
