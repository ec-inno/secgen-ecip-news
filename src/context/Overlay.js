import React, { useContext } from 'react';

const Context = React.createContext();

const useOverlayContext = () => useContext(Context);

export { useOverlayContext, Context as default };
