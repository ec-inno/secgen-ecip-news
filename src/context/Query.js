import React, { useContext } from 'react';

import { defaultLangKey } from '../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'unsetFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

const useQueryContext = () => useContext(Context);

export { queryInit, queryReducer, Context as default, useQueryContext };
