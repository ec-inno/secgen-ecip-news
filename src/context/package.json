{
  "name": "@eci/context",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@ecl/eu-preset-website": "2.8.0",
    "react": "16.9.0"
  }
}
