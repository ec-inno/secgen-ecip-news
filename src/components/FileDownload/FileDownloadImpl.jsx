import React from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import mime from 'mime';

import { useI18nContext } from '@eci/context/I18n';
import formatBytes from '@eci/utils/formatBytes';
import { useDocumentApi, getResource } from '@eci/utils/useDocumentApi';
import getLanguages from '@eci/utils/getLanguages';

import Link from '../Link/LinkGatsby';
import ErrorMessage from '../ErrorMessage';
import { FileDownload } from './FileDownload';

const FileDownloadImpl = ({ file }) => {
  if (!file || Object.keys(file).length === 0) return '';
  if (!file.id && (!file.otherLanguages || file.otherLanguages.length === 0))
    return '';

  const { location } = useI18nContext();
  const { t } = useTranslation();
  const langsT = getLanguages(t);

  const { otherLanguages, language: fileLanguage } = file;

  const langT = langsT.find(l => l.value === fileLanguage);
  const language = langT ? langT.label : '';

  const languages = otherLanguages
    ? otherLanguages.map(language => {
        let urlPath = '';
        let pathParts = [];

        const translation = langsT.find(l => l.value === language);

        pathParts = location.pathname.split('/').filter(p => p);

        pathParts.shift();

        urlPath = pathParts.join('/');
        urlPath = `/${translation.value}/${urlPath}/`;

        if (location.hash) {
          urlPath += location.hash;
        }

        // Link attributes for the component's spread.
        return { label: translation.label, href: urlPath };
      })
    : [];

  if (!file.id && languages.length) {
    return (
      <p className="ecl-u-type-paragraph">
        {t('Other languages available')}:{' '}
        {languages.map((attr, key) => [
          key > 0 && ', ',
          <Link {...attr} key={key} />,
        ])}
      </p>
    );
  }

  const { error } = useDocumentApi(file.id);
  const resource = getResource(file.id);

  let meta = '';

  const title = file.name ? file.name : '';

  if (file.size) {
    meta += formatBytes(file.size);
  }

  if (file.mimeType) {
    meta += ' - ';
    meta += mime.getExtension(file.mimeType).toUpperCase();
  }

  if (Object.keys(error).length !== 0) {
    return <ErrorMessage title={t('File service error')} error={error} />;
  }

  return (
    <FileDownload
      title={title}
      language={language}
      meta={meta}
      download={{ label: t('Download'), href: resource }}
      icon={{ shape: 'general--copy', size: '2xl' }}
    />
  );
};

FileDownloadImpl.propTypes = {
  file: PropTypes.shape({
    name: PropTypes.string,
    language: PropTypes.string,
    otherLanguages: PropTypes.arrayOf(PropTypes.string),
    mimeType: PropTypes.string,
    id: PropTypes.number,
    size: PropTypes.number,
  }),
};

export default FileDownloadImpl;
