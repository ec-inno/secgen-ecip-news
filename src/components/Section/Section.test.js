import React from 'react';
import renderer from 'react-test-renderer';

import * as I18nContext from '@eci/context/I18n';

import FileDownload from '../FileDownload';
import Section from './Section';

describe('Section', () => {
  it('returns an empty string when no input', () => {
    const tree = renderer.create(<Section />).toJSON();
    expect(tree).toBe('');
  });

  it('returns an empty string if only title is provided', () => {
    const tree = renderer.create(<Section title="Section title" />).toJSON();
    expect(tree).toBe('');
  });

  it('can render children even if no title is provided', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
      location: {
        href: 'http://localhost:8000/en/initiatives/#1',
        pathname: '/en/initiatives/',
        hash: '#1',
      },
    }));

    const tree = renderer
      .create(
        <Section>
          <FileDownload
            file={{
              name: 'test',
              mimeType: 'text/plain',
              id: 16806,
              size: 10,
            }}
          />
        </Section>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
