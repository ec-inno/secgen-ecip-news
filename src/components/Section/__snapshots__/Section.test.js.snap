// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section can render children even if no title is provided 1`] = `
<div
  className="ecl-file"
  data-ecl-file={true}
>
  <div
    className="ecl-file__container"
  >
    <svg
      aria-hidden="true"
      className="ecl-file__icon ecl-icon ecl-icon--2xl ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--copy"
      />
    </svg>
    <div
      className="ecl-file__info"
    >
      <div
        className="ecl-file__title"
      >
        test
      </div>
      <div
        className="ecl-file__meta"
      >
        (
        10 Bytes
         -
         
        TXT
        )
      </div>
    </div>
    <a
      className="ecl-link ecl-link--standalone"
      download={true}
      href="undefined/register/document/16806"
    >
      <span
        className="ecl-link__label"
      >
        Download
      </span>
       
      <svg
        aria-hidden="true"
        className="ecl-icon--fluid ecl-link__icon ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#ui--download"
        />
      </svg>
    </a>
  </div>
</div>
`;
